### PR TITLE
[ready for review] allow sage.repl.interpreter._do_preparse to override preparse argument

### PR DESCRIPTION
**Pull request by DrXyzzy**
Proposed fix for #399 is to override `Salvus.execute()` argument `preparse` if `preparser(False)` has been called.

Test cases are in https://cloud.sagemath.com/projects/ccd4d4a4-29a8-4c39-85c2-a630cb1e9b6c/files/SAGEWS/issue399test.sagews. That project is running the proposed modification.

### CODE REVIEW

**Comment by haraldschilly** [comment]
So, I think that's a good fix. The only thing that comes to my mind is, what if the Sage API changes this in a year? Maybe wrap that import in a `try catch` like:

```
try:
    import sage.repl.interpreter as sage_repl_interpreter
except:
    # warning
```
and then
```
getattr(sage_repl_interpreter, '_do_preparse', True)
```
in the if.

**Comment by DrXyzzy** [comment]
Yes, of course, will add try/catch & update PR.

File: src/smc_sagews/smc_sagews/sage_server.py [hunk 1]
```diff
@@ -929,8 +929,18 @@ def execute(self, code, namespace=None, preparse=True, locals=None):
...
+        try:
+            import sage.repl.interpreter as sage_repl_interpreter
+        except:
+            log("Error - unable to import sage.repl.interpreter")
+
         for start, stop, block in blocks:
-            if preparse:
+            # if import sage.repl.interpreter fails, sag_repl_interpreter is unreferenced
+            try:
+                do_pp = getattr(sage_repl_interpreter, '_do_preparse', True)
```
Comment (haraldschilly) [inline src/smc_sagews/smc_sagews/sage_server.py]: oh! you don't need to do that try/catch here. `getattr` has **3** arguments here.

**Comment by DrXyzzy** [comment]
try/catch still needed because getattr generates NameError exception if first argument is not defined.

**Pull request merged**